### ISSUE TITLE
Rotation.GetEulerZYX parameters not wrapped

### ISSUE THREAD

**Issue by tahsintasdemir** [slater1/Kdl.NetStandard#3]
Hello there 

I am trying to apply the example here but there are some problems. =>  https://orocos.org/kdl/examples.html
1) I cannot add items to JntArray => jointpositions[0] = Jointangles[0] * Math.PI / 180;

2) cartpos.M.GetEulerZYX(alpha, beta, gamma); =>how do I define SWIGTYPE_p_double for alpha beta and gamma.

Thanks in advance..

            // Create a KDL chain object
            Chain chain = new Chain();

            // Add segments to the chain
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotZ), new Frame(new Vector(0.0, 0.0, 330.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotY), new Frame(new Vector(0.0, 0.0, 440.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotY), new Frame(new Vector(420.0, 0.0, 50.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotX), new Frame(new Vector(0.0, 0.0, 0.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotY), new Frame(new Vector(80.0, 0.0, 0.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotZ), new Frame(new Vector(0.0, 0.0, 0.0))));

            // Create the forward kinematics solver
            var fk = new ChainFkSolverPos_recursive(chain);

            var nj = chain.getNrOfJoints();
            JntArray jointpositions = new JntArray(nj);
            jointpositions[0] = Jointangles[0] * Math.PI / 180;
            jointpositions[1] = Jointangles[0] * Math.PI / 180;
            jointpositions[2] = Jointangles[0] * Math.PI / 180;
            jointpositions[3] = Jointangles[0] * Math.PI / 180;
            jointpositions[4] = Jointangles[0] * Math.PI / 180;
            jointpositions[5] = Jointangles[0] * Math.PI / 180;

            Frame cartpos = new Frame();
            var kinematics_status = fk.JntToCart(jointpositions, cartpos);

            var x = cartpos.p.x();
            var y = cartpos.p.y();
            var z = cartpos.p.z();

            cartpos.M.GetEulerZYX(alfa, beta, gamma);

**Comment by slater1** [comment]
Hey! Thanks for opening this issue.

I took some shortcuts when making this wrapper. For example, I didn't use `Rotation.GetEulerZYX`, so it still has funky wrapping (`double&` => `SWIGTYPE_p_double`).

I just now pushed a fix, and I bumped the NuGet package to [2.0.6](https://www.nuget.org/packages/Kdl.NetStandard.x64/2.0.6).

I added some tests based on your code:

```cs
    [Test]
    public void GetEulerZyx_IsCorrect()
    {
        var jointAngles = new double[] { 0, Math.PI/2, Math.PI/8, 0, 0, 0 };
        JntArray jointpositions = new((uint)jointAngles.Length);

        foreach (int i in Enumerable.Range(0, 6))
        {
            jointpositions.set(i, jointAngles[i]);
        }

        Frame cartesianPos = new();

        // Create the forward kinematics solver
        var fk = new ChainFkSolverPos_recursive(chain_);
        var kinematics_status = fk.JntToCart(jointpositions, cartesianPos);
        Assert.AreEqual(0, kinematics_status);

        cartesianPos.M.GetEulerZYX(out double alfa, out double beta, out double gamma);

        Assert.AreEqual(Math.PI, alfa);
        Assert.AreEqual(Math.PI/4 + Math.PI/8, beta);
        Assert.AreEqual(Math.PI, gamma);
    }
```

I hope this fixes it for you. Let me know how it goes!

**Comment by tahsintasdemir** [comment]
Hello,

I applied forward and inverse kinematics and started to get the results I wanted. I'm exploring the library now.
I will get back to you if I encounter any problems.

Thanks for the package.


            // Create a KDL chain object
            Chain chain = new Chain();

            // Add segments to the chain
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotZ), new Frame(new Vector(50.0, 0.0, 330.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotY), new Frame(new Vector(0.0, 0.0, 440.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotY), new Frame(new Vector(420.0, 0.0, 35.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotX), new Frame(new Vector(0.0, 0.0, 0.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotY), new Frame(new Vector(80.0, 0.0, 0.0))));
            chain.addSegment(new Segment(new Joint(Joint.JointType.RotZ), new Frame(new Vector(0.0, 0.0, 0.0))));


            // Create the forward kinematics solver
            var fksolver = new ChainFkSolverPos_recursive(chain);

            var nj = chain.getNrOfJoints();
            JntArray jointpositions = new JntArray(nj);
            foreach (int i in Enumerable.Range(0, 6))
            {
                jointpositions.set(i, Jointangles[i + 1] * Math.PI / 180);
            }


            Frame cartesianPos = new Frame();
            var kinematics_status = fksolver.JntToCart(jointpositions, cartesianPos);

            double x = 0, y = 0, z = 0, w = 0, p = 0, r = 0;
            if (kinematics_status >= 0)
            {
                x = cartesianPos.p.x();
                y = cartesianPos.p.y();
                z = cartesianPos.p.z();

                cartesianPos.M.GetRPY(out double alfa, out double beta, out double gamma);
                w = alfa * 180 / Math.PI;
                p = beta * 180 / Math.PI;
                r = gamma * 180 / Math.PI;
            }

          

            //Creation of jntarrays:
            JntArray q = new JntArray(chain.getNrOfJoints());
            JntArray q_init = new JntArray(chain.getNrOfJoints());
            JntArray q_min = new JntArray(chain.getNrOfJoints());
            JntArray q_max = new JntArray(chain.getNrOfJoints());
            

            foreach (int i in Enumerable.Range(0, 6))
            {
                q_init.set(i, 0 * Math.PI / 180);
            }

            foreach (int i in Enumerable.Range(0, 6))
            {
                q_min.set(i, Joints[i + 1].limitmin * Math.PI / 180);
            }

            foreach (int i in Enumerable.Range(0, 6))
            {
                q_max.set(i, Joints[i + 1].limitmax * Math.PI / 180);
            }


            //Creation of the ik solvers:
            var iksolverv = new ChainIkSolverVel_pinv(chain);//Inverse velocity solver
            var iksolver = new ChainIkSolverPos_NR_JL(chain, q_min, q_max, fksolver, iksolverv, 1000, 1e-6);//Newton-Raphson-Maximum 100 iterations, stop at accuracy 1e-6


            //Set destination frame
           
            Frame F_dest = new Frame(new Rotation(Rotation.RPY(w * Math.PI / 180, p * Math.PI / 180, r * Math.PI / 180)), new Vector(cartesianPos.p.x(), cartesianPos.p.y(), cartesianPos.p.z()));

            int ret = iksolver.CartToJnt(q_init, F_dest, q);
            string strerror;

            if (ret >= 0)
            {
                double j1 = q.get(0) * 180 / Math.PI;
                double j2 = q.get(1) * 180 / Math.PI;
                double j3 = q.get(2) * 180 / Math.PI;
                double j4 = q.get(3) * 180 / Math.PI;
                double j5 = q.get(4) * 180 / Math.PI;
                double j6 = q.get(5) * 180 / Math.PI;
      

            }
            else
                strerror = iksolver.strError(ret);
